### PR TITLE
feat: search toggles in project

### PR DESCRIPTION
**Pull request by ykhedher**
Screen Shot 2022-02-28 at 4 29 15 PM

### CODE REVIEW

**Comment by ykhedher** [comment]
[Frontend - Project page search](https://trello.com/c/pI4VyN5g/900-frontend-project-page-search)

**Comment by ykhedher** [comment]
[project search](https://trello.com/c/bbFsqn5w/913-project-search)

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.styles.ts [hunk 1]
```diff
@@ -37,4 +37,17 @@ export const useStyles = makeStyles(theme => ({
...
         textDecoration: 'none',
     },
+    actionsContainer: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    search: {
+        border: `1px solid ${theme.palette.grey[300]}`,
+        height: '35px',
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.styles.ts:47]: `35`

File: src/component/common/SearchField/styles.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const useStyles = makeStyles(theme => ({
     },
     searchIcon: {
         marginRight: '8px',
```
Comment (olav) [inline src/component/common/SearchField/styles.js:22]: `px` values can be numbers:

```
marginRight: 8,
```

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 3]
```diff
@@ -1,36 +1,47 @@
...
     loading,
 }: IProjectFeatureToggles) => {
     const styles = useStyles();
-    const { id } = useParams();
+    const { id } = useParams<{ id: string }>();
     const history = useHistory();
     const { hasAccess } = useContext(AccessContext);
     const { uiConfig } = useUiConfig();
+    const [filter, setFilter] = useState('');
+    useState<IFeatureToggleListItem[]>(features);
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx:37]: Looks unused.

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 4]
```diff
@@ -1,36 +1,47 @@
...
+
+    const filteredFeatures = useMemo(() => {
+        const regExp = new RegExp(filter, 'i');
+        return filter
+            ? features?.filter(feature => regExp.test(feature?.name))
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx:42]: Looks like `features` will always be defined, so we can drop the `?`. Same for `feature.name`.

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 5]
```diff
@@ -70,19 +88,20 @@ const ProjectFeatureToggles = ({
...
                             </ResponsiveButton>
-                        </>
+                        </div>
                     }
                 />
             }
         >
             <ConditionallyRender
-                condition={features?.length > 0}
+                condition={filteredFeatures?.length > 0}
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx:101]: Can remove `?` here as well.

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 6]
```diff
@@ -113,5 +132,3 @@ const ProjectFeatureToggles = ({
         </PageContent>
     );
 };
-
-export default ProjectFeatureToggles;
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx:117]: :clap:

File: src/component/common/ResponsiveButton/ResponsiveButton.tsx [hunk 7]
```diff
@@ -24,6 +25,7 @@ const ResponsiveButton: React.FC<IResponsiveButtonProps> = ({
     permission,
     environmentId,
     projectId,
+    className,
```
Comment (olav) [inline src/component/common/ResponsiveButton/ResponsiveButton.tsx:28]: Looks unused.

File: src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 8]
```diff
@@ -1,36 +1,47 @@
...
+import { IFeatureToggleListItem } from 'interfaces/featureToggle';
+import { getCreateTogglePath } from 'utils/route-path-helpers';
 import { useStyles } from './ProjectFeatureToggles.styles';
-import { CREATE_FEATURE } from '../../../providers/AccessProvider/permissions';
-import useUiConfig from '../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { CREATE_FEATURE } from 'component/providers/AccessProvider/permissions';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import classnames from 'classnames';
 
 interface IProjectFeatureToggles {
```
Comment (olav) [inline src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx:22]: `IProjectFeatureTogglesProps`

**Review by olav — CHANGES_REQUESTED** [review]

**Review by olav — APPROVED** [review]

**Pull request merged**